nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PublishSlack.java

```diff
@@ -0,0 +1,516 @@
+            .displayName("Channel")
+            .description("Slack channel, private group, or IM channel to post the message to.")
+            .dependsOn(PUBLISH_STRATEGY, MESSAGE_ONLY, CONTENT_MESSAGE, UPLOAD_ATTACH, UPLOAD_LINK)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor UPLOAD_CHANNEL = new PropertyDescriptor.Builder()
+            .name("upload-channel")
+            .displayName("Upload Channel")
+            .description("Slack channel, private group, or IM channel to upload the file to.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor THREAD_TS = new PropertyDescriptor.Builder()
+            .name("Thread Timestamp")
+            .description("The timestamp of the parent message, also known as a thread_ts, or Thread Timestamp. If not specified, the message will be send to the channel " +
+                    "as an independent message. If this value is populated, the message will be sent as a reply to the message whose timestamp is equal to the given value.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TEXT = new PropertyDescriptor.Builder()
+            .name("text")
+            .displayName("Text")
+            .description("Text of the Slack message to send. Only required if no attachment has been specified and 'Upload File' has been set to 'No'.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor INITIAL_COMMENT = new PropertyDescriptor.Builder()
+            .name("upload-text")
+            .displayName("Upload Text")
+            .description("Text of the Slack message to include with a public upload. Only used if 'Publish Mode' has been set to 'Upload Only' and 'Upload Channel' is specified.")
+            .dependsOn(PUBLISH_STRATEGY, UPLOAD_ONLY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor FILE_TITLE = new PropertyDescriptor.Builder()
+            .name("file-title")
+            .displayName("File Title")
+            .description("Title of the file displayed in the Slack message." +
+                    " The property value will only be used if 'Upload FlowFile' has been set to 'Yes'.")
```

Comment:
Not sure that I understand the description here. There is nowhere for the user to set 'Upload FlowFile'.